lib/_http_outgoing.js

```diff
@@ -511,11 +532,24 @@ OutgoingMessage.prototype.end = function(data, encoding, callback) {
     throw new TypeError('first argument must be a string or Buffer');
   }
 
+  var self = this;
+
   if (this.finished) {
+    if (data && data.length > 0) {
+      //If the user tried to write data, tell him it failed
```

Comment:
Style nit: please put a space after the slashes and use punctuation.  I've been told some people take offense at gendered pronouns in comments.

I'm not sure how I feel about turning write-after-end into an error.  It's arguably the proper thing to do but I worry that it's going to break a lot of existing code.